packages/tooling/fast-tooling/src/data-utilities/shortcuts.spec.md

```diff
@@ -0,0 +1,119 @@
+# Abstract
+
+The shortcuts utility accepts an instance of the `MessageSystem` and through this sends information about available shortcuts and events.
+
+## Instantiating
+
+All the `Shortcuts` class needs to instantiate is an instance of the `MessageSystem`. It should then fire a custom event notifying all other items registered to the message system that shortcuts are available, with a listener that will capture keypress and identify which shortcut action to execute.
+
+```typescript
+import { Shortcut } from "@microsoft/fast-tooling"
+
+new Shortcuts({
+    messageSystem: fastMessageSystem
+});
+```
+
+## Adding shortcut actions
+
+Shortcut actions can either be custom or imported from a set of actions available.
+
+```typescript
+import { DuplicateShortcutAction, ShortcutAction, Shortcuts } from "@microsoft/fast-tooling"
+
+new Shortcuts({
+    messageSystem: fastMessageSystem,
+    actions: [
+        // Custom shortcut
+        new ShortcutAction({
+            name: "Random action",
+            keys: [
+                {
+                    altKey: true
+                },
+                {
+                    value: "Tab"
+                }
+            ],
+            action: () => {
```

Comment:
Is there a way to get an action by name and invoke it manually?

Reply:
I've added an `id` to the supplied config that can then be used to identify and run an action that has been supplied.